replicatedhq/ship
# Gracefully fail if a Helm chart has missing values

What I Did
------------
- Return 400 on `saveHelmValues` if linting of Helm templates fail
- Update UI to halt continuing if a Helm lint error is detected

How I Did it
------------
Incorporate Helm's linter as a dependency in the project, and run against Values passed from frontend in the `helm-values` route. I also updated the UI to properly halt and display a 400 error if it detects a failure.

How to verify it
------------
Try skipping values with this command:
```
ship init https://github.com/sourcegraph/deploy-sourcegraph 
```
It should fail with the error below:
```
[ERROR] templates/: render error in \"sourcegraph/templates/storage-class.yaml\": template: sourcegraph/templates/storage-class.yaml:2:27: executing \"sourcegraph/templates/storage-class.yaml\" at <required \".Values.cl...>: error calling required: .Values.cluster.storageClass.create is required
```
Description for the Changelog
------------
- Gracefully fail if a Helm chart has missing values


Picture of a Boat (not required but encouraged)
------------
![](https://vignette.wikia.nocookie.net/avatar/images/1/14/Steamship.png/revision/latest?cb=20131128023456)
